sys/kern/kern_malloc.c

```diff
@@ -437,6 +442,11 @@ malloc_type_freed(struct malloc_type *mtp, unsigned long size)
 	mtsp = zpcpu_get(mtip->mti_stats);
 	mtsp->mts_memfreed += size;
 	mtsp->mts_numfrees++;
+#ifdef __CHERI_PURE_CAPABILITY__
+	mtsp->mts_memunreserved += cheri_getlen(addr);
```

Comment:
It's probably worth a comment about the assumption in the commit message somewhere. Eventually we're going to need to look at hardening the `free` path, but that's for another day.

Reply:
Added note in `malloc_type_freed()` comment.